### PR TITLE
Get JIT orders by tx

### PR DESCRIPTION
**Pull request by sunce86**
# Description
A follow up for https://github.com/cowprotocol/services/pull/2920

Implements [https://api.cow.fi/mainnet/api/v1/transactions/{tx_hash}/orders](https://api.cow.fi/docs/#/default/get_api_v1_transactions__txHash__orders) for JIT orders.

# Changes

- [ ] implemented `get_by_tx` database function to retrieve all jit orders in a transaction
- [ ] expanded orderbook api function `orders_for_tx` to return jit orders as well

## How to test
updated e2e test

### CODE REVIEW

File: crates/e2e/tests/e2e/jit_orders.rs [hunk 1]
```diff
@@ -193,9 +193,17 @@ async fn single_limit_order_test(web3: Web3) {
...
-    assert_eq!(orders.len(), 1);
+    // jit order can be found on /api/v1/transactions/{tx_hash}/orders
+    let orders_by_tx = services
+        .get_orders_for_tx(&orders[0].tx_hash.unwrap())
+        .await
+        .unwrap();
+    assert!(orders_by_tx
+        .iter()
+        .map(|o| o.metadata.uid)
+        .any(|uid| uid == jit_order_uid));
```
Comment (sunce86) [inline crates/e2e/tests/e2e/jit_orders.rs:208]: proof that it works

File: crates/database/src/jit_orders.rs [hunk 2]
```diff
@@ -16,19 +16,24 @@ use {
...
+const JIT_ORDERS_SELECT: &str = r#"
+o.uid, o.owner, o.creation_timestamp, o.sell_token, o.buy_token, o.sell_amount, o.buy_amount,
+o.valid_to, o.app_data, o.fee_amount, o.kind, o.partially_fillable, o.signature,
+o.receiver, o.signing_scheme, o.sell_token_balance, o.buy_token_balance,
+(SELECT COALESCE(SUM(t.buy_amount), 0) FROM trades t WHERE t.order_uid = o.uid) AS sum_buy,
+(SELECT COALESCE(SUM(t.sell_amount), 0) FROM trades t WHERE t.order_uid = o.uid) AS sum_sell,
+(SELECT COALESCE(SUM(t.fee_amount), 0) FROM trades t WHERE t.order_uid = o.uid) AS sum_fee,
+COALESCE((SELECT SUM(surplus_fee) FROM order_execution oe WHERE oe.order_uid = o.uid), 0) as executed_surplus_fee
+"#;
+
```
Comment (sunce86) [inline crates/database/src/jit_orders.rs:28]: just moved / extracted, no code change

File: crates/e2e/tests/e2e/jit_orders.rs [hunk 3]
```diff
@@ -193,9 +193,14 @@ async fn single_limit_order_test(web3: Web3) {
...
-    // jit order can be found on /get_trades
+    // jit order can be found on /api/v1/trades
     let orders = services.get_trades(&jit_order_uid).await.unwrap();
-    assert_eq!(orders.len(), 1);
+    // jit order can be found on /api/v1/transactions/{tx_hash}/orders
+    let orders_by_tx = services
+        .get_orders_for_tx(&orders[0].tx_hash.unwrap())
+        .await
+        .unwrap();
+    assert!(orders_by_tx.iter().any(|o| o.metadata.uid == jit_order_uid));
```
Comment (sunce86) [inline crates/e2e/tests/e2e/jit_orders.rs:205]: proof that it works

File: crates/database/src/jit_orders.rs [hunk 4]
```diff
@@ -39,6 +44,29 @@ COALESCE((SELECT SUM(surplus_fee) FROM order_execution oe WHERE oe.order_uid = o
...
+        orders::SETTLEMENT_LOG_INDICES,
+        "SELECT ",
+        JIT_ORDERS_SELECT,
+        " FROM jit_orders o 
+        JOIN trades t ON t.order_uid = o.uid",
+        " WHERE
+        t.block_number = (SELECT block_number FROM settlement) AND
+        -- BETWEEN is inclusive
+        t.log_index BETWEEN (SELECT * from previous_settlement) AND (SELECT log_index FROM \
+         settlement) ",
```
Comment (sunce86) [inline crates/database/src/jit_orders.rs:61]: copy paste from same functionality https://github.com/cowprotocol/services/blob/main/crates/database/src/orders.rs#L605-L609

File: crates/orderbook/src/database/orders.rs [hunk 5]
```diff
@@ -329,13 +329,22 @@ impl OrderStoring for Postgres {
...
             .map(|result| match result {
                 Ok(order) => full_order_into_model_order(order),
                 Err(err) => Err(anyhow::Error::from(err)),
             })
             .try_collect()
             .await
+            .map(|mut orders: Vec<Order>| {
+                orders.extend(jit_orders);
+                orders
+            })
```
Comment (squadgazzz) [inline crates/orderbook/src/database/orders.rs:347]: Does it make sense to slightly speed up the response time by executing these queries in parallel?

**Review by squadgazzz — APPROVED** [review]
LGTM

**Review by fleupold — APPROVED** [review]

File: crates/orderbook/src/database/orders.rs (hunk 5 above)
Comment (sunce86) [reply crates/orderbook/src/database/orders.rs]: Implemented

**Pull request merged**